csharp/src/Ice/Proxy.cs

```diff
@@ -415,21 +415,36 @@ public static class Proxy
                                                                progressWrapper,
                                                                cancel).ConfigureAwait(false);
 
-                            switch (response.ReplyStatus)
+                            // TODO: the observer RemoteException notification is not quite correct, since with the 1.1
+                            // encoding it suppresses (on purpose) the system exceptions.
+                            if ((response.Encoding == Encoding.V1_1 &&
+                                    response.ReplyStatus == ReplyStatus.UserException) ||
+                                (response.Encoding != Encoding.V1_1 && response.ResultType == ResultType.Failure))
                             {
-                                case ReplyStatus.OK:
-                                    break;
-                                case ReplyStatus.UserException:
-                                    observer?.RemoteException();
-                                    break;
-                                case ReplyStatus.ObjectNotExistException:
-                                case ReplyStatus.FacetNotExistException:
-                                case ReplyStatus.OperationNotExistException:
-                                    throw response.ReadDispatchException();
-                                case ReplyStatus.UnknownException:
-                                case ReplyStatus.UnknownLocalException:
-                                case ReplyStatus.UnknownUserException:
-                                    throw response.ReadUnhandledException();
+                                observer?.RemoteException();
```

Comment:
The simplest would be too call it as well for 1.1 system exceptions... having two different behaviour depending on the encoding sounds too complicated and confusing for users.

Reply:
Yes, fixed.